File: selfstudy/python/stars_filterlow.py
'''
task 186:
Stars - FilterLow
You need to return only the values less than 50 from star luminosities
'''

def below_fifty_stars(brightness):
    return [x for x in brightness if x < 50]

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], below_fifty_stars([10, 20, 30])),
        ([50, 60, 70], below_fifty_stars([50, 60, 70])),
        ([1, 2, 3], below_fifty_stars([1, 2, 3])),
        ([100, 200], below_fifty_stars([100, 200])),
        ([], below_fifty_stars([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = below_fifty_stars(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# done

File: selfstudy/js/stars_filterlow.js
/*
task 86:
Stars - FilterLow
You need to return only the values less than 50 from star luminosities
*/

function below_fifty_stars(brightness) {
  return brightness.filter(x => x < 50);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: below_fifty_stars([10, 20, 30]) },
  { input: [50, 60, 70], expected: below_fifty_stars([50, 60, 70]) },
  { input: [1, 2, 3], expected: below_fifty_stars([1, 2, 3]) },
  { input: [100, 200], expected: below_fifty_stars([100, 200]) },
  { input: [], expected: below_fifty_stars([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('below_fifty_stars', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(below_fifty_stars(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, below_fifty_stars(t.input) === t.expected || JSON.stringify(below_fifty_stars(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works
